vivisect/analysis/elf/elfplt.py

```diff
@@ -14,61 +14,43 @@ def analyze(vw):
     Do simple linear disassembly of the .plt section if present.
     """
     for sva,ssize,sname,sfname in vw.getSegments():
-        if sname != ".plt":
+        if sname not in (".plt", ".plt.got"):
             continue
+
         nextva = sva + ssize
         while sva < nextva:
             vw.makeCode(sva)
             ltup = vw.getLocation(sva)
             sva += ltup[vivisect.L_SIZE]
 
 def analyzeFunction(vw, funcva):
-
     seg = vw.getSegment(funcva)
     if seg == None:
         return
 
     segva, segsize, segname, segfname = seg
 
-    if segname != ".plt":
+    if segname not in (".plt", ".plt.got"):
         return
 
-    #FIXME check for i386
     op = vw.parseOpcode(funcva)
-    if op.opcode != opcode86.INS_BRANCH:
+    if op.iflags & envi.IF_BRANCH == 0:
         return
 
-    loctup = None
     oper0 = op.opers[0]
+    opval = oper0.getOperAddr(op, None)
 
-    if isinstance(oper0, e_i386.i386ImmMemOper):
-        
-        loctup = vw.getLocation(oper0.getOperAddr(op))
-
-    elif isinstance(oper0, e_i386.i386RegMemOper):
-        # FIXME this is i386 elf only!
-        if oper0.reg != e_i386.REG_EBX:
-            print "UNKNOWN PLT CALL",hex(funcva)
-        got = vw.vaByName("%s._GLOBAL_OFFSET_TABLE_" % segfname)
-
-        #FIXME this totally sucks
-        if got == None:
-            for va,size,name,fname in vw.getSegments():
-                if name == ".got.plt":
-                    got = va
-                    break
-
-        if got != None:
-            loctup = vw.getLocation(got+oper0.disp)
+    loctup = vw.getLocation(opval)
 
     if loctup == None:
         return
 
-    if loctup[vivisect.L_LTYPE] != vivisect.LOC_IMPORT:
+    if loctup[vivisect.L_LTYPE] != vivisect.LOC_IMPORT: # FIXME: Why are some AMD64 IMPORTS showing up as LOC_POINTERs?
```

Comment:
It's even some 32 bit pointers. 
Basic example I can make is `__gmon_start__` which gets compiled in. My guess is that it shows up as a basic pointer because it's just a null pointer in the GOT (since I didn't compile w/ `-pg` or `-g`). But I'll dive into the ELF parser to see why and circle back to this.

Reply:
This could also be some kinda weird order of operations issue where the import is not constructed before the brute-force-pointer-finder phase ( maybe because it gets missed or because it fails to make it? )

Reply:
found it.  it seems like a few different issues.  one: Imports can only be Imports if they have a name associated with them.  some of these don't seem to.  also, i'm seeing a bunch of R_X86_64_* reloc types we don't support (namely R_X86_64_IRELATIV, which i'm having a hard time tracking down.  if they're not marked as *something*, codeflow analysis marks them as pointers:
```
              elif bflags & envi.BR_DEREF:
  
                  self.addXref(va, tova, REF_DATA)
                  ptrdest = None
                  if self.getLocation(tova) is None:
                      ptrdest = self.makePointer(tova, follow=False)
```
which of course makes it into a pointer, since it's not already marked as an Import.

Reply:
https://refspecs.linuxfoundation.org/elf/x86_64-abi-0.99.pdf  reading up.

Reply:
improvements  :)  
now i have to go retool the unit tests.